### PR TITLE
Curve whispering

### PR DESCRIPTION
**Pull request by 0xJepsen**
This PR finished the g2 curve interface.

- Added tests
- Squashed some bugs

It starts on the setup part of KZG to generate the SRS. The last thing we need to do for this is to support a `from` for biguints.

### CODE REVIEW

**Comment by devloper** [comment]
simply unmatched. I think i'ma ronk with ya'll tomorrow.

**Review by devloper — COMMENTED** [review]
nice!

File: math/field.sage [hunk 1]
```diff
@@ -66,15 +66,16 @@ print(omega_n, "^", n, " = ", omega_n ^ n)
 # extension field of degree 2
 Ft.<t> = F[]
 
-# irreducible  element: t^2-2
-P = Ft(t ^ 2 - 2)
+# irreducible  element: t^2 + 2
+P = Ft(t ^ 2 + 2)
```
[the remark this answers is not in the document]
Comment (0xJepsen) [reply math/field.sage:70]: There are some algorithms to do it! Essentially an irreducible element here means that it cannot be factored in our base field.  Wheres x^ + 1 = (x+ 10)(x - 10) in our base field.  That is a counter example!

[the remark this answers is not in the document]
Comment (Autoparallel) [reply math/field.sage:70]: Galois tells us how

**Review by Autoparallel — APPROVED** [review]

**Pull request merged**